VirtoCommerce.Storefront/Domain/Security/CustomCookieAuthenticationEvents.cs

```diff
@@ -25,8 +33,23 @@ public override Task RedirectToAccessDenied(RedirectContext<CookieAuthentication
                 context.Response.StatusCode = (int)HttpStatusCode.Forbidden;
                 return Task.CompletedTask;
             }
+            context.RedirectUri = AdjustRedirectUri(context.RedirectUri);
             return base.RedirectToAccessDenied(context);
         }
-        
+
+        /// <summary>
+        /// add store and lng segments of current store to uri
+        /// </summary>
+        /// <param name="uri"></param>
+        /// <returns></returns>
+        private string AdjustRedirectUri(string uri)
```

Comment:
Rename to `AddStoreAndLanguageToUri` and remove XML documentation.

For future:
- Don't cut words (lng > language)
- URI in comments should be uppercase